src/components/Canvas.js

```diff
@@ -0,0 +1,305 @@
+				drawDragAnchor(img.x + img.width, img.y, ctx)
+				drawDragAnchor(img.x, img.y + img.height, ctx)
+				drawDragAnchor(img.x + img.width, img.y + img.height, ctx)
+				// drawDragAnchor(img.x + img.width/2, img.y + img.height, ctx)
+			}
+		}
+	}
+
+	function anchorHitTest(x, y, imgIndex) {
+		let dx, dy
+		let img = images[imgIndex]
+		// top-left
+		dx = x - img.x
+		dy = y - img.y
+		if (dx * dx + dy * dy <= rr) {
+			return 0
+		}
+		// top-right
+		dx = x - (img.x + img.width)
+		dy = y - img.y
+		if (dx * dx + dy * dy <= rr) {
+			return 1
+		}
+		// bottom-right
+		dx = x - (img.x + img.width)
+		dy = y - (img.y + img.height)
+		if (dx * dx + dy * dy <= rr) {
+			return 2
+		}
+		// bottom-left
+		dx = x - img.x
+		dy = y - (img.y + img.height)
+		if (dx * dx + dy * dy <= rr) {
+			return 3
+		}
+
+		// dx = x - (img.x + img.width/2)
+		// dy = y - (img.y + img.height)
+		// if (dx * dx + dy * dy <= rr) {
+		// 	return 4
+		// }
+
+		return -1
+	}
+
+	const drawDragAnchor = (x, y, ctx) => {
+		ctx.beginPath()
+		ctx.arc(x, y, resizerRadius, 0, pi2, false)
+		ctx.closePath()
+		ctx.fill()
+	}
+
+	// test if x,y is inside the bounding box of texts[textIndex]
+	const imageHitTest = (x, y, imgIndex) => {
+		let img = images[imgIndex]
+		return (
+			x >= img.x &&
+			x <= img.x + img.width &&
+			y <= img.y + img.height &&
+			y >= img.y
+		)
+	}
+
+	const handleMouseDown = e => {
+		if(!e._reactName.includes("Touch")) {
+			e.preventDefault()
+			setStartX(parseInt(e.clientX - offsetX))
+			setStartY(parseInt(e.pageY - offsetY))
+		} else {
+			setStartX(parseInt(e.changedTouches[0].clientX - offsetX))
+			setStartY(parseInt(e.changedTouches[0].pageY - offsetY))
+		}
+
+		// Put your mousedown stuff here
+		for (let i = 1; i < images.length; i++) {
+			const corner = anchorHitTest(startX, startY, i)
+			if (corner > -1) {
+				console.log('hit anchor', corner)
+
+				if(corner === 4) {
+					setRotatingImg(corner)
+				} 
+				setDraggingResizer(corner)
+				setSelectedImg(i)
+			} else if (imageHitTest(startX, startY, i)) {
+				console.log('hit image')
+				setSelectedImg(i)
+				if (rotating === i) {
+					setRotatingImg(-1)
+				} else {
+					setRotatingImg(i)
+				}
+			}
+		}
+	}
+
+	const handleMouseMove = e => {
+		let mouseX;
+		let mouseY;
+		if (!e._reactName.includes('Touch')) {
+			e.preventDefault()
+			mouseX = parseInt(e.clientX - offsetX)
+			mouseY = parseInt(e.pageY - offsetY)
+		} else {
+			mouseX = parseInt(e.changedTouches[0].clientX - offsetX)
+			mouseY = parseInt(e.changedTouches[0].pageY - offsetY)
+		}
+		let img = images[selectedImg]
+		setStartX(mouseX)
+		setStartY(mouseY)
+		if(!img) {
+			return
+		}
+
+		if (selectedImg < 1) {
+			return
+		}
+
+		// Put your mousemove stuff here
+		let dx = mouseX - startX
+		let dy = mouseY - startY
+
+		if (draggingResizer > -1) {
+			// resize the image
+			const imageRight = img.x + img.width
+			const imageBottom = img.y + img.height
+			switch (draggingResizer) {
```

Comment:
for `switch` statements it's usually a good idea to put a `default` case - just to catch any weird outliers that may come up if something gets changed elsewhere!

a simple warning like this would be fine (just to avoid silent errors):
```
    // ^ the expected cases...
default:
    console.warn('Canvas.js - received an unexpected value for draggingResizer')
```